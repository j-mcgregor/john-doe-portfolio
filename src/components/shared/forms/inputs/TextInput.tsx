import React from 'react'
import { TextInputProps } from '../../../../types/interfaces'

// We could use const TextInput: React.FC<React.InputHTMLAttributes<HTMLInputElement>>
// But since this is a composite component including a label, we'll do a custom interface

const TextInput: React.FC<TextInputProps> = ({
    type,
    placeholder,
    label,
    name,
    required,
    value,
    onChange,
    classNames = [],
}) => (
    <div className={`form-group ${classNames.join(' ')}`}>
        {label && <label>{label}</label>}
        <input
            type={type}
            placeholder={placeholder}
            className="form-control"
            name={name}
            required={required}
            value={value}
            onChange={onChange}
        />
    </div>
)

export default TextInput
