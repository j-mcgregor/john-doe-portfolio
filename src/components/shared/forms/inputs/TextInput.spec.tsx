import React from 'react'
import { render, fireEvent } from '@testing-library/react'
import TextInput from './TextInput'

describe('TextInput', () => {
    it('should render an input', () => {
        const mockFn = jest.fn()
        const utils = render(
            <TextInput
                type="text"
                label="Test Text Input"
                placeholder="Test"
                name="test"
                value=""
                onChange={mockFn}
                required
            />
        )

        expect(utils.baseElement).toMatchSnapshot()
    })

    it('should trigger the onChange by changing the input', () => {
        const mockFn = jest.fn()
        const utils = render(
            <TextInput
                type="text"
                label="Test Text Input"
                placeholder="Test"
                name="test"
                value=""
                onChange={mockFn}
                required
            />
        )

        const input = utils.getByLabelText('test')
        fireEvent.change(input, { target: { value: '123' } })
        expect(mockFn).toHaveBeenCalledTimes(1)
    })
})
