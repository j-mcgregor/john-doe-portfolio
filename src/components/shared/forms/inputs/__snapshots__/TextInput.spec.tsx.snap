// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render an input 1`] = `
<body>
  <div>
    <div
      class="form-group "
    >
      <label
        for="test"
      >
        Test Text Input
      </label>
      <input
        aria-label="test"
        class="form-control"
        data-testid="test"
        name="test"
        placeholder="Test"
        required=""
        type="text"
        value=""
      />
    </div>
  </div>
</body>
`;
